recipe/recipe.go

```diff
@@ -1,38 +1,121 @@
 package recipe
 
+import (
+	"fmt"
+	"strconv"
+)
+
 //List - struct for list of recipes
 type List struct {
-	Count      int        `bson:"count,omitempty" json:"count"`
-	Offset     int        `bson:"offset_index,omitempty" json:"offset_index"`
-	Limit      int        `bson:"limit,omitempty" json:"limit"`
-	Items      []Response `bson:"items,omitempty" json:"items"`
-	TotalCount int        `bson:"total_count,omitempty" json:"total_count"`
+	Count      int        `bson:"count" json:"count"`
+	Offset     int        `bson:"offset_index" json:"offset_index"`
+	Limit      int        `bson:"limit" json:"limit"`
+	Items      []Response `bson:"items" json:"items"`
+	TotalCount int        `bson:"total_count" json:"total_count"`
 }
 
 //Response - struct for individual recipe
 type Response struct {
 	ID              string     `bson:"_id" json:"id"`
-	Alias           string     `bson:"alias,omitempty" json:"alias,omitempty"`
-	Format          string     `bson:"format,omitempty" json:"format,omitempty"`
-	InputFiles      []file     `bson:"files,omitempty" json:"files,omitempty"`
-	OutputInstances []instance `bson:"output_instances,omitempty" json:"output_instances,omitempty"`
+	Alias           string     `bson:"alias" json:"alias"`
+	Format          string     `bson:"format" json:"format"`
+	InputFiles      []file     `bson:"files" json:"files"`
+	OutputInstances []instance `bson:"output_instances" json:"output_instances"`
 }
 
 //CodeList - Code lists for instance
 type CodeList struct {
-	ID          string `bson:"id,omitempty" json:"id,omitempty"`
-	HRef        string `bson:"href,omitempty" json:"href,omitempty"`
-	Name        string `bson:"name,omitempty" json:"name,omitempty"`
-	IsHierarchy bool   `bson:"is_hierarchy,omitempty" json:"is_hierarchy,omitempty"`
+	ID          string `bson:"id" json:"id"`
+	HRef        string `bson:"href" json:"href"`
+	Name        string `bson:"name" json:"name"`
+	IsHierarchy *bool  `bson:"is_hierarchy" json:"is_hierarchy"`
 }
 
 type instance struct {
-	DatasetID string     `bson:"dataset_id,omitempty" json:"dataset_id,omitempty"`
-	Editions  []string   `bson:"editions,omitempty" json:"editions,omitempty"`
-	Title     string     `bson:"title,omitempty" json:"title,omitempty"`
-	CodeLists []CodeList `bson:"code_lists,omitempty" json:"code_lists,omitempty"`
+	DatasetID string     `bson:"dataset_id" json:"dataset_id"`
+	Editions  []string   `bson:"editions" json:"editions"`
+	Title     string     `bson:"title" json:"title"`
+	CodeLists []CodeList `bson:"code_lists" json:"code_lists"`
 }
 
 type file struct {
-	Description string `bson:"description,omitempty" json:"description,omitempty"`
+	Description string `bson:"description" json:"description"`
+}
+
+//Validate - checks if all the fields are non-empty
+func (recipe *Response) Validate() error {
```

Comment:
Missing unit tests for Validate function (I would expect to see recipe_test.go file)

Reply:
Done in commit "Add validation to endpoints and unit tests"